cmd/zc_filter.go

```diff
@@ -156,3 +174,26 @@ func buildIncludeFilters(patterns []string) []objectFilter {
 
 	return []objectFilter{&includeFilter{patterns: validPatterns}}
 }
+
+type filterSet []objectFilter
+
+// GetEnumerationPreFilter returns a prefix that is common to all the include filters, or "" if no such prefix can
+// be found. (The implementation may return "" even in cases where such a prefix does exist, but in at least the simplest
+// cases, it should return a non-empty prefix.)
+// The result can be used to optimize enumeration, since anything without this prefix will fail the filterSet
+func (fs filterSet) GetEnumerationPreFilter() string {
+	prefix := ""
+	for _, f := range fs {
+		if participatingFilter, ok := f.(preFilterProvider); ok {
+			// this filter knows how to participate in our scheme
+			if prefix == "" {
+				prefix = participatingFilter.getEnumerationPreFilter()
+			} else {
+				// prefix already has a value, which means there must be two participating filters, and we can't handle that.
```

Comment:
I'm also feeling a bit nitpicky about this. If we can find a good system to select a common root prefilter in the individual prefilters, then we can do it here.

So, if the above comment gets changes, this one should too.

Reply:
This is something we could do in 10.4.  I don't think we should do it in this PR, for the reasons outlined in my other answer.